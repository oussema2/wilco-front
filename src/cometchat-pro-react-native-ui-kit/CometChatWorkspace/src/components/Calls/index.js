export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
