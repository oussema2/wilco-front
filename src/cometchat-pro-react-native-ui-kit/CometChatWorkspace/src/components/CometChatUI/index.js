/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
