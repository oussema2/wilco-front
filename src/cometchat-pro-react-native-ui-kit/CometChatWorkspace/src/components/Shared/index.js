export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
