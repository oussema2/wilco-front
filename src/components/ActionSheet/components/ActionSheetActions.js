import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
