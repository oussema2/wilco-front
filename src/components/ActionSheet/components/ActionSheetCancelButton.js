import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
