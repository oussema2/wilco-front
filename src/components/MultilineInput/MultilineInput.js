import React from 'react';
import {
	View, TextInput as InputElement, Text
} from 'react-native';
import PropTypes from 'prop-types';
import {
	viewStyles
} from './MultilineInput.styles';
import { useFocus } from '../../hooks/useFocus';
import { palette } from '../../Theme';
import { useInputsStyle } from '../../hooks/useInputsStyle';
import { exclamationTriangle } from '../../assets/icons';
import { Image } from '../Image';

const MultilineInput = ( {
	testID, value, error, disabled, onFocus, onBlur, onChange, minimumLines, capitalizeFirstLetter,
	placeholder, maxLength, bold, inputRef
} ) => {
	const [ focused, onInputFocused, onInputBlurred ] = useFocus( onFocus, onBlur );
	const [ styles, minHeight ] = useInputsStyle(
		focused, error, disabled, minimumLines, value, bold
	);

	const maxLengthExceeded = value?.length > maxLength;

	const backgroundColor = ( error || maxLengthExceeded )
		? palette.error.background : styles.input.backgroundColor;

	return (
		<View style={viewStyles.container}>
			<InputElement
				testID={testID}
				style={{
					...styles.input,
					minHeight,
					backgroundColor
				}}
				ref={inputRef}
				placeholderTextColor={palette.grayscale.aluminum}
				selectionColor={palette.primary.default}
				value={value}
				editable={!disabled}
				onFocus={onInputFocused}
				onBlur={onInputBlurred}
				onChangeText={onChange}
				placeholder={placeholder}
				autoCapitalize={( capitalizeFirstLetter ) ? 'sentences' : 'none'}
				multiline
				scrollEnabled={false} // necessary for KeyboardAvoidingView to work
			/>

			{!!maxLength && ( !error || maxLengthExceeded ) && (
				<View style={styles.footnoteContainer}>
					{maxLengthExceeded && (
						<Image testID="alert-icon" source={exclamationTriangle} style={styles.footnoteIcon} resizeMode="contain" />
					)}
					<Text style={maxLengthExceeded ? styles.footnoteError : styles.footnote} align="right">{`${value?.length || 0}`}</Text>
					<Text style={styles.footnote}>{`/${maxLength}`}</Text>
				</View>
			)}

			{!!error && !maxLengthExceeded && (
				<Text testID="helperText-testID" style={styles.errorMessage}>
					{error}
				</Text>
			)}
		</View>
	);
};

MultilineInput.propTypes = {
	testID: PropTypes.string,
	value: PropTypes.string,
	error: PropTypes.string,
	disabled: PropTypes.bool,
	onFocus: PropTypes.func,
	onBlur: PropTypes.func,
	onChange: PropTypes.func,
	minimumLines: PropTypes.number,
	capitalizeFirstLetter: PropTypes.bool,
	placeholder: PropTypes.string,
	maxLength: PropTypes.number,
	inputRef: PropTypes.oneOfType( [
		PropTypes.func,
		PropTypes.shape( { current: PropTypes.any } )
	] ),
	bold: PropTypes.bool
};

MultilineInput.defaultProps = {
	testID: 'MultilineInput-Component',
	value: '',
	error: '',
	disabled: false,
	onFocus: () => {},
	onBlur: () => {},
	onChange: () => {},
	minimumLines: 1,
	capitalizeFirstLetter: false,
	placeholder: '',
	maxLength: undefined,
	inputRef: undefined,
	bold: true
};

export default MultilineInput;
