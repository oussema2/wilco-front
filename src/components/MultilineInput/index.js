export { default as MultilineInput } from './MultilineInput';
