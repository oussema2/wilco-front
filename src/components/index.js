export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
