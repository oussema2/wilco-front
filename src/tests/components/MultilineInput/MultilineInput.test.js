import React from 'react';
import { render } from '@testing-library/react-native';
import { MultilineInput } from '../../../components/MultilineInput';

describe( '<MultilineInput />', () => {
	let component;
	const testID = 'testing-MultilineInput-Component';

	const setUp = () => {
		component = render( <MultilineInput testID={testID} /> );
	};

	beforeEach( () => {
		setUp();
	} );

	describe( 'with the default props', () => {
		it( 'renders the MultilineInput component correctly', () => {
			expect( component.queryByTestId( testID ) ).toBeDefined();

			expect( component ).toMatchSnapshot();
		} );
	} );
} );
