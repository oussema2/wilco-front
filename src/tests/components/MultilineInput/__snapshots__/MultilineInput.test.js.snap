// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultilineInput /> with the default props renders the MultilineInput component correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TextInput
    autoCapitalize="none"
    editable={true}
    multiline={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder=""
    placeholderTextColor="#A0A0A0"
    scrollEnabled={false}
    selectionColor="#12A3BE"
    style={
      Object {
        "backgroundColor": "#F2F2F2",
        "borderRadius": 8,
        "color": "#000000",
        "fontFamily": "NotoSansJP-Bold",
        "fontSize": 14,
        "fontWeight": "normal",
        "letterSpacing": 0,
        "lineHeight": 20,
        "minHeight": 48,
        "paddingBottom": 16,
        "paddingHorizontal": 12,
        "paddingTop": 16,
        "textAlignVertical": "top",
      }
    }
    testID="testing-MultilineInput-Component"
    value=""
  />
</View>
`;
